src/bloom.cpp

```diff
@@ -197,3 +208,38 @@ void CBloomFilter::UpdateEmptyFull()
     isFull = full;
     isEmpty = empty;
 }
+
+CRollingBloomFilter::CRollingBloomFilter(unsigned int nElements, double fpRate, unsigned int nTweak) :
+    b1(nElements*2, fpRate, nTweak), b2(nElements*2, fpRate, nTweak)
+{
+    // Implemented using two bloom filters of nElements each.
+    // We fill them up, and clear them, staggered, every nElements
+    // inserted, so at least one always contains the last nElements
+    // inserted.
+    nBloomSize = nElements*2;
+    nInsertions = 0;
+}
+
+void CRollingBloomFilter::insert(const std::vector<unsigned char>& vKey)
+{
+    if (nInsertions%nBloomSize == 0)
+        b1.clear();
+    else if (nInsertions%nBloomSize == nBloomSize/2)
+        b2.clear();
+    b1.insert(vKey);
+    b2.insert(vKey);
+    ++nInsertions;
```

Comment:
I'd prefer to wrap this counter around at nBloomSize, this avoids some `%` operations, and also avoids trouble with integer overflows when UIINT_MAX is not divisible by nBloomSize.